docassemble/ALToolbox/business_days.py

```diff
@@ -120,64 +121,52 @@ def non_business_days(
 
 
 def is_business_day(
-    date, country="US", subdiv="MA", add_holidays=None, remove_holidays=None
+    date: Union[str, DADateTime],
+    country="US",
+    subdiv="MA",
+    add_holidays=None,
+    remove_holidays=None,
 ) -> bool:
-    if (
-        date
-        in non_business_days(
-            as_datetime(date).year,
-            country=country,
-            subdiv=subdiv,
-            add_holidays=add_holidays,
-            remove_holidays=remove_holidays,
-        ).keys()
+    if not isinstance(date, DADateTime):
+        date = as_datetime(date)
+    if date.dow in [
+        6,
+        7,
+    ]:  # Docassemble codes Saturday and Sunday as 6 and 7 respectively
+        return False
+    if date.format("yyyy-MM-dd") in standard_holidays(
```

Comment:
`standard_holidays` returns date objects, not strings, so the string will never be in the list of holidays. You'd have to change back to `non_business_days` to get it working again.

Reply:
standard_holidays returns a dictionary, where the key is an ISO formatted date string.

Reply:
Hm, that's a weird class. [This function](https://github.com/dr-prodigy/python-holidays/blob/122387489d4a7a98b58d873c4412f58f0e0f1469/holidays/holiday_base.py#L262-L289) is the critical function, it stores everything internally as a datetime obj, and transforms strings to the comparable datetime when using it. I was getting really confused yesterday when I was trying it in the REPL (see below). I guess my tests were wrong somehow when I thought it was broken.

TBH I'd keep the type as `dict`, and either remove the doc string you added, or change it to say it uses either strings or date objects as keys.

```
>>> from docassemble.ALToolbox.business_days import standard_holidays
>>> standard_holidays(2022)
{datetime.date(2022, 1, 1): "New Year's Day", datetime.date(2022, 1, 17): 'Martin Luther King Jr. Day', datetime.date(2022, 2, 21): "Washington's Birthday", datetime.date(2022, 4, 18): "Patriots' Day", datetime.date(2022, 5, 30): 'Memorial Day', datetime.date(2022, 6, 19): 'Juneteenth National Independence Day', datetime.date(2022, 6, 20): 'Juneteenth National Independence Day (Observed)', datetime.date(2022, 7, 4): 'Independence Day', datetime.date(2022, 9, 5): 'Labor Day', datetime.date(2022, 10, 10): 'Columbus Day', datetime.date(2022, 11, 11): 'Veterans Day', datetime.date(2022, 11, 24): 'Thanksgiving', datetime.date(2022, 12, 25): 'Christmas Day', datetime.date(2022, 12, 26): 'Christmas Day (Observed)'}
>>> 
```